src/main/java/io/pinecone/configs/PineconeConfig.java

```diff
@@ -0,0 +1,63 @@
+package io.pinecone.configs;
+
+import io.grpc.ManagedChannel;
+import io.pinecone.exceptions.PineconeValidationException;
+
+public class PineconeConfig {
+
+    private String apiKey;
+    private String host;
+    private String usageContext;
+    private ManagedChannel customChannelBuilder;
+
+    public PineconeConfig(String apiKey) {
+        this.apiKey = apiKey;
+    }
+
+    public String getApiKey() {
+        return apiKey;
+    }
+
+    public void setApiKey(String apiKey) {
+        this.apiKey = apiKey;
+    }
+
+    public String getHost() {
+        return host;
+    }
+
+    public void setHost(String host) {
+        this.host = host;
+    }
+
+    public String getUsageContext() {
+        return usageContext;
+    }
+
+    public void setUsageContext(String usageContext) {
+        this.usageContext = usageContext;
+    }
+
+    public ManagedChannel getCustomChannelBuilder() {
+        return this.customChannelBuilder;
+    }
+
+    public void setCustomChannelBuilder(ManagedChannel customChannelBuilder) {
+        this.customChannelBuilder = customChannelBuilder;
+    }
+
+    public interface CustomChannelBuilder {
+        ManagedChannel buildChannel();
+    }
+
+    void validate() {
+        if (apiKey == null)
+            throw new PineconeValidationException("Invalid PineconeConfig: missing apiKey");
+    }
+
+    public String getUserAgent() {
+        String userAgentLanguage = "lang=java; pineconeClientVersion = v0.8.0";
```

Comment:
We shouldn't be hard-coding this version string like this. We will definitely forget to update it when releasing.

Reply:
We shouldn't be hardcoding and we already have a ticket for this. We used to extract the sdk version at runtime but we had a bug so going to tackle this after the java reference docs. 

As a part of release process, it is one of the manual steps that I follow every time I release. In the README file, we also have to update it manually which is something I'll like to tackle later.